### ISSUE TITLE
[REF-2162] Radix rx.radio(["1","2","3"]) fails with unsupported operand type

### ISSUE THREAD

**Issue by vhunyad** [reflex-dev/reflex#2747]
**Describe the bug**
File "/srv/tracker/venv/lib/python3.9/site-packages/reflex/components/radix/themes/components/radio_group.py", line 158, in create
    def radio_group_item(value: str | Var) -> Component:
TypeError: unsupported operand type(s) for |: 'type' and 'type'

However, the rx.radio.root+item version works.

**To Reproduce**
`rx.radio(["1","2","3"])`

**Specifics (please complete the following information):**
 - Python Version: 3.9.2
 - Reflex Version: 0.4.2
 - OS: debian 11
 

<sub>[REF-2162](https://linear.app/reflex-dev/issue/REF-2162/radix-rxradio[123]-fails-with-unsupported-operand-type)</sub>

**Comment by ElijahAhianyo** [comment]
Can you give an example code to repro this ?

**Comment by vhunyad** [comment]
There is nothing special, as simple as that:

```
def comp() -> rx.Component:
    return rx.radio(["1","2","3"])
```

**Comment by masenf** [comment]
needs `from __future__ import annotations` in this file `reflex/components/radix/themes/components/radio_group.py`.

@Lendemor we should probably make sure that all the new modules have this to ensure py39 / py39 compat

**Comment by ericwb** [comment]
This appears to be already fixed as part of PR #2841
